extensions/amp-auto-ads/0.1/test/test-ad-strategy.js

```diff
@@ -0,0 +1,207 @@
+/**
+ * Copyright 2016 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+
+import {AdStrategy} from '../ad-strategy';
+import {PlacementState, getPlacementsFromConfigObj} from '../placement';
+import * as sinon from 'sinon';
+
+describe('amp-auto-ads', () => {
+
+  let sandbox;
+  let container;
+
+  beforeEach(() => {
+    sandbox = sinon.sandbox.create();
+    container = document.createElement('div');
+    document.body.appendChild(container);
+  });
+
+  afterEach(() => {
+    sandbox.restore();
+    document.body.removeChild(container);
+  });
+
+  it('should place an ad in the first placement only with correct attributes',
+      done => {
+        const anchor1 = document.createElement('div');
+        anchor1.id = 'anId1';
+        container.appendChild(anchor1);
+
+        const anchor2 = document.createElement('div');
+        anchor2.id = 'anId2';
+        container.appendChild(anchor2);
+
+        const configObj = {
+          placements: [
+            {
+              anchor: {
+                selector: 'DIV#anId1',
+              },
+              pos: 2,
+              type: 1,
+            },
+            {
+              anchor: {
+                selector: 'DIV#anId2',
+              },
+              pos: 2,
+              type: 1,
+            },
+          ],
+        };
+        const placements = getPlacementsFromConfigObj(window, configObj);
+        expect(placements.length).to.equal(2);
+
+        const adStrategy = new AdStrategy('ad-network-type', placements, [
+          {
+            name: 'custom-att-1',
+            value: 'val-1',
+          },
+          {
+            name: 'custom-att-2',
+            value: 'val-2',
+          },
+        ]);
+
+        adStrategy.run().then(success => {
+          expect(success).to.equal(true);
+          expect(anchor1.childNodes.length).to.equal(1);
+          expect(anchor2.childNodes.length).to.equal(0);
+          const adElement = anchor1.childNodes[0];
+          expect(adElement.tagName).to.equal('AMP-AD');
+          expect(adElement.getAttribute('type')).to.equal('ad-network-type');
+          expect(adElement.getAttribute('data-custom-att-1')).to.equal('val-1');
+          expect(adElement.getAttribute('data-custom-att-2')).to.equal('val-2');
+          done();
```

Comment:
you can omit the `done()` by return the whole promise. you'll also need to remove `done` from the `it` method.